sdk/test-utils/perfstress/src/testProxyHttpClient.ts

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { HttpOperationResponse, URLBuilder } from "@azure/core-http";
+import { DefaultHttpClient, WebResource, WebResourceLike } from "@azure/core-http";
+import {
+  PipelinePolicy,
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest,
+  createDefaultHttpClient,
+  HttpClient,
+  createPipelineRequest
+} from "@azure/core-rest-pipeline";
+
+const paths = {
+  playback: "/playback",
+  record: "/record",
+  start: "/start",
+  stop: "/stop"
+};
+
+export class TestProxyHttpClientV1 extends DefaultHttpClient {
+  private _uri: string;
+  private _httpClient: DefaultHttpClient;
+  private _recordingId?: string;
+  public _mode!: string;
+
+  constructor(uri: string) {
+    super();
+    this._uri = uri;
+    this._httpClient = new DefaultHttpClient();
+  }
+
+  async sendRequest(request: WebResourceLike): Promise<HttpOperationResponse> {
+    if (this._recordingId && (this._mode === "record" || this._mode === "playback")) {
+      request.headers.set("x-recording-id", this._recordingId);
+      request.headers.set("x-recording-mode", this._mode);
+      request.headers.set("x-recording-remove", "false");
+
+      const upstreamUrl = URLBuilder.parse(request.url);
+      const redirectedUrl: URLBuilder = URLBuilder.parse(request.url);
+
+      const providedUrl = URLBuilder.parse(this._uri);
+      redirectedUrl.setHost(providedUrl.getHost());
+      redirectedUrl.setScheme(providedUrl.getScheme());
+      redirectedUrl.setPort(providedUrl.getPort());
+      upstreamUrl.setPath(undefined);
+      request.headers.set("x-recording-upstream-base-uri", upstreamUrl.toString());
+      request.url = redirectedUrl.toString();
+    }
+
+    return await super.sendRequest(request);
+  }
+
+  async startRecording(): Promise<void> {
```

Comment:
should we try to make this class robust to state changes? Like what happens if `startRecording` is called a second time before `stopRecording`?

Reply:
These methods won't be called by the users, only meant for the framework to call, so there is very little need to guard.
I can add something nonetheless. :)

Reply:
Added a state manager